chalicelib/action.py

```diff
@@ -98,18 +99,35 @@ def _add_action(self):
             self.send_response(message=message)
             return ""
 
-        # begin validate date range
-        if date == "today":
-            date = datetime.now().strftime(self.format_str)
+        date = parse_date(date=date, format_str=self.format_str)
+
+        if not date:
+            self.send_response(message="failed to parse date {date}")
 
-        if not validate_date(date, self.format_str):
-            self.send_response(message="failed to parse date")
+        date_list = list(date.values())
+        first_date = date_list[0]
+        # second date is optional
+        second_date = None
+        if len(date_list) > 1:
+            second_date = date_list[1]
 
-        self.send_attachment(
-            attachment=submit_message_menu(self.user_name, reason, date, hours)
+        dates_to_check = get_dates(
+            first_date=first_date,
+            second_date=second_date,
         )
```

Comment:
Does this section make sense to you guys?

Reply:
We're really checking months here?

Reply:
@tommymalmqvist Yes, `dates_to_check` will be a list of all the dates that we want to check if locked. Any suggestion on how to make it more clear?